### PR TITLE
Hooks: mpl_toolkits.basemap: Add conda specific places to check for auxiliary data.

### PR DESCRIPTION
**Pull request by giumas**
added additional places to check for auxiliary data (conda specific)

### CODE REVIEW

File: PyInstaller/hooks/hook-mpl_toolkits.basemap.py [hunk 1]
```diff
@@ -8,9 +8,32 @@
...
 # mpl_toolkits.basemap (tested with v.1.0.7) is shipped with auxiliary data,
 # usually stored in mpl_toolkits\basemap\data and used to plot maps
-datas = collect_data_files('mpl_toolkits.basemap')
+datas = collect_data_files('mpl_toolkits.basemap', subdir='data')
+
+# check if the data has been effectively found
+if len(datas) == 0:
+    
+    # - conda-specific
+    if hasattr(sys, 'real_prefix'):  # check if in a virtual environment
```
Comment (htgoebel) [inline PyInstaller/hooks/hook-mpl_toolkits.basemap.py]: In `PyInstaller.compat` we are using a more sophisticated method to get the `base_prefix`, so we can handle different kinds of virtual environments. Any reason yu are not using `PyInstaller.compat.base_prefix`? Did you test your code with conda + virtualenv, resp. conada + pyvenv?

**Comment by htgoebel** [comment]
When updating the commit, please also change the commit message into something like: "Hooks: mpl_toolkits.basemap: Add conda specific places to check for auxiliary data." (but mind the length restriction).

When updating a pull-request, you can simply (force) push the updated branch to github again. This will automatically update the pull-request (which follows the branch, not the commit). So you do not need to close the pull-request and open a new one. This also has the benefit that the discussion history is kept.

File: PyInstaller/hooks/hook-mpl_toolkits.basemap.py (hunk 1 above)
Comment (giumas) [inline PyInstaller/hooks/hook-mpl_toolkits.basemap.py]: To be honest, I was not aware of `PyInstaller.compat.base_prefix`!

**Comment by htgoebel** [comment]
This pull-request now ha 11 commit. Can you please rebase your changes? Thanks. Like this.
git checkout  'update_basemap_hook'
git rebase develop
git push -f update_basemap_hook'

File: PyInstaller/hooks/hook-mpl_toolkits.basemap.py (hunk 1 above)
Comment (htgoebel) [inline PyInstaller/hooks/hook-mpl_toolkits.basemap.py]: Nevermind :-)

**Comment by giumas** [comment]
Sorry for the delayed action! I have just tested it on a Conda env and it works.

**Comment by xoviat** [comment]
This patch [appears to cause][1] a test failure.

[1]: https://travis-ci.org/pyinstaller/pyinstaller/jobs/197098087#L1781

**Comment by giumas** [comment]
@xoviat It fails, but I don't understand the reason based on the minimal changes that this patch applies. 

There are a couple of `ImportError`s for `queue` and `urllib3` on travis-ci. Are these the issues?

**Comment by xoviat** [comment]
I removed my comment because I realized that I was not correct.

**Comment by giumas** [comment]
@htgoebel Do I need to change something else on this PR?

**Comment by htgoebel** [comment]
@giumas Thanks for triggering :-)

**Pull request merged**